crawl-ref/source/spl-util.cc

```diff
@@ -1344,8 +1344,12 @@ int spell_highlight_by_utility(spell_type spell, int default_colour,
     {
         return COL_FORBIDDEN;
     }
+    // Grey out spells for which you lack experience or spell levels.
+    if (spell_difficulty(spell) > you.experience_level
+             || player_spell_levels() < spell_levels_required(spell))
+        default_colour = DARKGRAY;
```

Comment:
I would use one of the semantic color constants in `spl-util.h` (or add a new one) rather than an actual color term.  `COL_INAPPLICABLE` might be appropriate, and is `DARKGRAY` anyways.

Also, this line should be in `{}` because the condition is two lines, I'm not sure why `unbrace` didn't catch it.